### ISSUE TITLE
Improve cancel button guidelines

### ISSUE THREAD

**Issue by maximedegreve** [primer/design#290]
Our [saving](https://primer.style/design/ui-patterns/saving#bottom-left-default) docs is already doing a great job at providing guidelines but one thing that's missing is more details on usage of a cancel button

One relevant bit on saving is:
```When data is not automatically saved after the user makes changes, buttons are used to submit or cancel the changes.```

This might imply we should always include a cancel button but that isn't correct.

- Should a `cancel` button be used on intermediate form pages (not just modals)
- Cancelling can be useful when by accident changing the form
- Are there accessibility considerations here?

[Original Slack thread (internal)](https://github.slack.com/archives/CPA865D9S/p1662734727898729)
